File: Leetcode/Daily/30-05-2024/prob.go
func countTriplets(arr []int) int {
    ans := 0
    n := len(arr)
    for i, val := range arr{
        for k := i + 1; k < n; k++{
            val ^= arr[k]
            if val == 0{
                ans += k - i
            }
        }
    }
    return ans
}

File: Leetcode/Daily/30-05-2024/prob.cpp
class Solution {
public:
    int countTriplets(vector<int>& arr) {
        arr.insert(arr.begin(),0);
        int n = arr.size();
        for(int i=1;i<n;i++)arr[i] ^= arr[i-1];

        int ans = 0;
        unordered_map<int,int>cnt,tot;
        for(int i=0;i<n;i++){
            ans += cnt[arr[i]]++ * (i-1)-tot[arr[i]];
            tot[arr[i]]+=i;
        }

        return ans;

    }
};